signal_test.go

```diff
@@ -0,0 +1,73 @@
+// Copyright (c) 2022 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package fx
+
+import (
+	"github.com/stretchr/testify/assert"
+	"syscall"
+	"testing"
+)
+
+func assertUnsentSignalError(
+	t *testing.T,
+	err error,
+	expected *unsentSignalError,
+) {
+	t.Helper()
+
+	actual := new(unsentSignalError)
+
+	assert.Error(t, err)
+	assert.ErrorContains(t, err, "channels are blocked")
```

Comment:
If you're doing ErrorContains, assert.Error is not needed because it'll check non-nil first.
```suggestion
	assert.ErrorContains(t, err, "channels are blocked")
```